src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -199,28 +199,50 @@ static void dev_urandom_close(void) {
 #if CRYPTOGRAPHY_OSRANDOM_ENGINE == CRYPTOGRAPHY_OSRANDOM_ENGINE_GETENTROPY
 static const char *Cryptography_osrandom_engine_name = "osrandom_engine getentropy()";
 
+static int getentropy_works = CRYPTOGRAPHY_OSRANDOM_GETENTROPY_NOT_INIT;
+
 static int osrandom_init(ENGINE *e) {
+#if !defined(__APPLE__)
+    getentropy_works = CRYPTOGRAPHY_OSRANDOM_GETENTROPY_WORKS;
+#else
+    if (getentropy != NULL) {
```

Comment:
This is the entire weak link check @alex. Combined with `-mmacosx-version-min=10.9` the linker knows that `getentropy` is only available on 10.12 and weak links it, so when running it on an older macOS version it will be a null.